src/precice/bindings/matlab/+precice/@SolverInterface/SolverInterface.m

```diff
@@ -0,0 +1,283 @@
+            if size(ids,2) ~= inSize
+                error('Number of columns in position vector must match size!');
+            end
+            positions = preciceGateway(uint8(47),int32(meshID),int32(inSize),vertexIds);
+        end
+        
+        % getMeshVertexIDsFromPositions
+        function vertexIds = getMeshVertexIDsFromPositions(obj,meshID,inSize,positions)
+            if size(positions,2) ~= inSize
+                error('Number of columns in position vector must match size!');
+            end
+            vertexIds = preciceGateway(uint8(48),int32(meshID),int32(inSize),positions);
+        end
+        
+        % setMeshEdge
+        function edgeID = setMeshEdge(obj, meshID, firstVertexID, secondVertexID)
+            edgeID = preciceGateway(uint8(49),int32(meshID),int32(firstVertexID),int32(secondVertexID));
+        end
+        
+        % setMeshTriangle
+        function setMeshTriangle(obj, meshID, firstEdgeID, secondEdgeID, thirdEdgeID)
+            preciceGateway(uint8(50),int32(meshID),int32(firstEdgeID),int32(secondEdgeID),int32(thirdEdgeID));
+        end
+        
+        % setMeshTriangleWithEdges
+        function setMeshTriangleWithEdges(obj, meshID, firstVertexID, secondVertexID, thirdVertexID)
+            preciceGateway(uint8(51),int32(meshID),int32(firstVertexID),int32(secondVertexID),int32(thirdVertexID));
+        end
+        
+        % setMeshQuad
+        function setMeshQuad(obj, meshID, firstEdgeID, secondEdgeID, thirdEdgeID, fourthEdgeID)
+            preciceGateway(uint8(52),int32(meshID),int32(firstEdgeID),int32(secondEdgeID),int32(thirdEdgeID),int32(fourthEdgeID));
+        end
+        
+        % setMeshQuadWithEdges
+        function setMeshQuadWithEdges(obj, meshID, firstVertexID, secondVertexID, thirdVertexID, fourthVertexID)
+            preciceGateway(uint8(53),int32(meshID),int32(firstVertexID),int32(secondVertexID),int32(thirdVertexID),int32(fourthVertexID));
+        end
+        
+        %% Data Access
+        % hasDataID
+        function bool = hasData(obj,dataName,meshID)
+            if ischar(dataName)
+                dataName = string(dataName);
+            end
+            bool = preciceGateway(uint8(60),dataName,int32(meshID));
+        end
+        
+        % getDataID
+        function id = getDataID(obj,dataName,meshID)
+            if ischar(dataName)
+                dataName = string(dataName);
+            end
+            id = preciceGateway(uint8(61),dataName,int32(meshID));
+        end
+        
+        % mapReadDataTo
+        function mapReadDataTo(obj,meshID)
+            preciceGateway(uint8(62),int32(meshID));
+        end
+        
+        % mapWriteDataFrom
+        function mapWriteDataFrom(obj,meshID)
+            preciceGateway(uint8(63),int32(meshID));
+        end
+        
+        % writeBlockVectorData
+        function writeBlockVectorData(obj,dataID,inSize,valueIndices,values)
```

Comment:
We could try to get rid of `inSize` by directly extracting the size from `size(valueIndices)`.